android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/dashboard/DashboardFragment.java

```diff
@@ -87,38 +102,82 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         }
     }
 
-    private void initializePieChart() {
+    private void populateCategories() {
         Context context = requireContext();
-        SortedSet<Cashflow> cashflows = ApplicationState.getInstance(context).getCashflows();
 
-        Stream<Cashflow> cashflowStream = cashflows.stream().filter(x -> CashflowType.EXPENSE == x.getType());
+        orderedCategoryList = ApplicationState.getInstance(context)
+                .getCashflows()
+                .stream()
+                .filter(x -> CashflowType.EXPENSE == x.getType())
+                .map(Cashflow::getCategory)
+                .collect(Collectors.toSet()) // make sure we have only one occurrence of each category
+                .stream() // then sort by localized name of the category
+                .sorted(Comparator.comparing(o -> context.getText(o.getName()).toString()))
+                .collect(Collectors.toList());
+    }
 
-        if (null != startDate) {
-            cashflowStream = cashflowStream.filter(x -> startDate.minusDays(1).isBefore(ChronoLocalDate.from(x.getTimestamp())));
-        }
+    private void initializeFilter() {
+        Button filterButton = binding.openFilterOptions;
+        filterButton.setOnClickListener(v -> {
+            Log.i("FiBo", "clicked filter button");
```

Comment:
Do you plan to keep these logs?

Reply:
Somewhere I saw another log (probably from an old implementation). Maybe you could also remove it